services/explorer/src/pages/Faucet/Faucet.effects.ts

```diff
@@ -0,0 +1,105 @@
+import { useSnackbar } from 'notistack';
+import { useCallback, useEffect, useState } from 'react';
+import { faucetUri } from '../../config';
+import { useExplorerNetwork } from '../../hooks/useExplorerNetwork';
+import { useExplorerLoader } from '../../hooks/useLoader';
+
+export enum RequestLimit {
+  ONE = 1,
+  TWO = 2,
+  FIVE = 5,
+  SEVEN = 7,
+}
+export const useFaucetEffects = () => {
+  const { currentNetwork } = useExplorerNetwork();
+  const { isLoading, initLoading, finishLoading } = useExplorerLoader();
+  const { enqueueSnackbar, closeSnackbar } = useSnackbar();
+  const [address, setAddress] = useState<string>('');
+  const [amount, setAmount] = useState<RequestLimit>(1);
+  const [displayToast, setDisplayToast] = useState<{
+    display: boolean;
+    variant?: 'success' | 'error' | 'warning' | undefined;
+    text?: string;
+  }>({ display: false, variant: undefined, text: '' });
+
+  const sendTokenRequest = useCallback(() => {
+    const requestTokens = async () => {
+      initLoading();
+      try {
+        const response = await fetch(faucetUri, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            address,
+            amount,
+            timestamp: Date.now() / 1000,
```

Comment:
We should ensure that address and amount are there so we don't do a request with empty or wrong values.
I think it might be good to add react-hook-form or something and yup for validating the address format.